redis/commands/search/hybrid_query.py

```diff
@@ -0,0 +1,326 @@
+        """
+        self._search_query = search_query
+        self._vector_similarity_query = vector_similarity_query
+
+    def get_args(self) -> List[str]:
+        args = []
+        args.extend(self._search_query.get_args())
+        args.extend(self._vector_similarity_query.get_args())
+        return args
+
+
+class HybridPostProcessingConfig:
+    def __init__(self) -> None:
+        """
+        Create a new hybrid post processing configuration object.
+        """
+        self._combine = []
+        self._load_fields = []
+        self._groupby = []
+        self._apply = []
+        self._sortby_fields = []
+        self._filter = None
+        self._limit = None
+
+    def combine(
+        self,
+        method: Literal["RRF", "LINEAR"],
+        **kwargs,
+    ) -> Self:
+        """
+        Add combine parameters to the query.
+
+        Args:
+            method: The combine method to use - RRF or LINEAR.
+            kwargs: Additional combine parameters.
+        """
+        self._combine: List[Union[str, int]] = [method]
+
+        self._combine.append(len(kwargs) * 2)
+
+        for key, value in kwargs.items():
+            self._combine.extend([key, value])
+
+        return self
+
+    def load(self, *fields: str) -> Self:
+        """
+        Add load parameters to the query.
+        """
+        self._load_fields = fields
+        return self
+
+    def group_by(self, fields: List[str], *reducers: Reducer) -> Self:
+        """
+        Specify by which fields to group the aggregation.
+
+        Args:
+            fields: Fields to group by. This can either be a single string or a list
+                of strings. In both cases, the field should be specified as `@field`.
+            reducers: One or more reducers. Reducers may be found in the
+                `aggregation` module.
+        """
+
+        fields = [fields] if isinstance(fields, str) else fields
+
+        ret = ["GROUPBY", str(len(fields)), *fields]
+        for reducer in reducers:
+            ret += ["REDUCE", reducer.NAME, str(len(reducer.args))]
+            ret.extend(reducer.args)
+            if reducer._alias is not None:
+                ret += ["AS", reducer._alias]
+
+        self._groupby.extend(ret)
+        return self
+
+    def apply(self, **kwexpr) -> Self:
+        """
+        Specify one or more projection expressions to add to each result.
+
+        Args:
+            kwexpr: One or more key-value pairs for a projection. The key is
+                the alias for the projection, and the value is the projection
+                expression itself, for example `apply(square_root="sqrt(@foo)")`.
+        """
+        for alias, expr in kwexpr.items():
+            ret = ["APPLY", expr]
+            if alias is not None:
+                ret += ["AS", alias]
+            self._apply.extend(ret)
+
+        return self
+
+    def sort_by(self, *sortby: "SortbyField") -> Self:
```

Comment:
Looks like `SortByField` is missing `WITHCOUNT` option

Reply:
This is in the design doc. I think it based on FT.SEARCH spec

https://redis.io/docs/latest/commands/ft.search/